### PR TITLE
Test repeat when evicted

### PR DESCRIPTION
**Pull request by PaulWoitaschek**
* Moved repeat logic to a separate transformer
* Test that transformer

### CODE REVIEW

**Comment by PaulWoitaschek** [comment]
Hm; what can I do to make git not show the merged commits?

**Comment by pavlospt** [comment]
@PaulWoitaschek How are you merging?

**Comment by digitalbuddha** [comment]
Instead of merging, you should rebase develop and then git push -f

**Comment by pavlospt** [comment]
Yeah that's true! Actually do a `git push` of your commit then `git rebase upstream/develop` and then `git push -f` your own develop, so that it will update the PR.

**Comment by PaulWoitaschek** [comment]
That worked

File: store/src/main/java/com/nytimes/android/external/store/base/impl/RepeatWhenEmits.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        this.source = source;
+    }
+
+    @Nonnull
+    static <T> RepeatWhenEmits<T> from(@Nonnull Observable source) {
+        return new RepeatWhenEmits<>(checkNotNull(source));
+    }
+
+    @Override
+    public Observable<T> call(Observable<T> upstream) {
```
Comment (digitalbuddha) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RepeatWhenEmits.java]: Still can barely understand this part. Thanks for writing test to make it clearer and for fixing my error.

Comment (PaulWoitaschek) [inline store/src/main/java/com/nytimes/android/external/store/base/impl/RepeatWhenEmits.java]: It was not your error, it was mine while renaming the signatures.

For understanding check [this post](http://blog.danlew.net/2016/01/25/rxjavas-repeatwhen-and-retrywhen-explained/) from Dan Lew.

**Pull request merged**